keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,635 @@
+and those are maintained with the help of sig-docs.
+
+#### Reference documentation
+
+The reference documentation update process needs changes.
+Currently tooling maintained by sig-docs does the following:
+1) Clones k/k
+2) Runs tools to generate cmd/kubeadm/app Cobra reference documentation
+3) Prettyfies the documentation with HTML/CSS
+
+The tooling needs to change to start generating the documentation
+from the k/kubeadm repository. SIG docs must be notified of this
+and collaborate with the kubeadm maintainers when the move is about to happen.
+
+There would not be a need for the tooling to continue building old
+reference documentation from k/k instead, as the k8s.io website reference
+documentation is only generated once before a release happens.
+
+The tools for building k/kubeadm reference docs in k/k must be removed.
+The initial proposal is to not support building reference documentation
+from the tools in the k/kubeadm repository, but might need some discussion.
+Currently the consumers of k/k reference documentation are unknown and the
+documentation is no longer tracked in commits. See:
+https://github.com/kubernetes/kubernetes/tree/master/docs
+
+#### Release notes
+
+The prow “release-note” plugin must be enabled for the kubeadm repository.
+This will make PRs require release notes or the option NONE. Once a k8s
+release has begun ideally the tool:
+https://github.com/kubernetes/release/tree/master/cmd/release-notes
+should be levereged to obtain all the release notes for kubeadm in this
+release.
+
+The alternative is for the kubeadm maintainers to collect the release
+notes manually and send them to the release team, but this might be more
+difficult to handle as the k8s release tooling sends emails with the release
+notes right away. Also the kubeadm maintainers might make mistakes forgetting
+about a certain feature/bug fix.
+
+### Test Plan
+
+#### CI artifacts
+
+Currently the kubeadm binary is built and uploaded in the CI GCS bucket
+next to the rest of the k8sbinaries like kubelet.
+
+There is a complication as now another repository has to be monitored
+for changes. Currently the CI build happens after a PR merged in the
+k/k repository.
+
+The proposal is to always rebuild kubeadm based on the latest tips of
+the master and release-x.yy branches of the k/kubeadm repository after
+a CI build for the k/k repository triggers. In this case the k/release
+build tooling must not trigger the writing of a KUBE_VERSION file in the
+k/kubeadm repository but rather let the version to generate based on the
+latest Git tags and SHAs.
+
+#### PR testing
+
+There is already support for running pre-submit jobs on k/kubeadm PRs.
+Once kubeadm is moved to the new location a collection of PR blocking
+jobs should be added to make sure the project is always buildable with
+the current version of Go that the k/k repository uses. Syncing Go versions
+should be the responsibility of the k/kubeadm maintainers.
```

Comment:
This may be sufficient short term, but we need to think of how to more actively manage this.  Perhaps some k/common that is branched with k/k and any other repos that follow k/k's release cycle (eg: k/release and as proposed in this KEP k/kubeadm).

Reply:
possibly yes. branching k/common will face similar issues as k/release, k/kubeadm or any other repository that wishes to follow the k/k versioning.

Reply:
Starts to argue for a tool like Android's `repo`....

You've argued it wont be a problem for kubeadm, but for other split out repos, if we have a "miss", and as you say "missing component will mean that the users need to build from source or wait for the next PATCH. the k8s train should not wait."...does the user end up needing two versions of golang?  Two versions of etcd?  And so on...?

I can accept that we ignore this for kubeadm.  But SIG Release will need to be thinking more about this and working for more active management of dependences across repos or we're going to end up at times with a set of components that are not truly capable of proper integration.

Reply:
i think that the tooling has to be re-written first, before working in this direction.

Reply:
This is another one that perhaps becomes an arch level convo, or needs contrib summit dialog around what ought to be common across the project and how to manage that.